pkg/graph/graph.go

```diff
@@ -22,6 +22,19 @@ func NewResourceManager() Manager {
 	}
 }
 
+func (rm *resourceManager) GetAllResource(ctx context.Context) []Resource {
+	resources := []Resource{}
+
+	rm.lock.RLock()
+	defer rm.lock.RUnlock()
+
+	for resource := range rm.resources {
+		resources = append(resources, resource)
```

Comment:
why we need to copy them from `rm.resources` to a new slice other than use or return `rm.resources` directly?

Reply:
Because the structure of this field is `map[Resource][]Resource`

Reply:
Thx, got it.